### DISCUSSION TITLE
TypeBox schema

### DISCUSSION THREAD

**Discussion by Ontopic** [payloadcms/payload#258] [Feature Requests & Ideas]
https://github.com/sinclairzx81/typebox

Sinnce the Typing is so well manicured and the JOI integration is quite similar... Worth considering to put in a plugin to generate a JS validator? Unless you guys have a better suggestion for a community plugin, I think I might give this a shot.

Could be used as a nice translation layer, to share configs from Payload with the frontend, even if they run in separate containers.

**Comment by DanRibbens** [comment]
I think what you're talking about is the same as this discussion https://github.com/payloadcms/payload/discussions/225. Is that fair to say?

That could be done as a plugin, but we prefer to have it in as part of the core library.

**Reply by Ontopic** [reply]
Completely missed that one, thanks, will continue there :) Although my main contribution probably is "typebox" 🙃

Have another plugin idea, will open a new discussion for that so you can close [do discussions work like that?] this one.
